### PR TITLE
Feature: Capture Errors with Sentry

### PR DESCRIPTION
**Pull request by jakeleboeuf**
# Describe your PR
Simply adds Sentry error logging and an Error Boundary.

Fixes [this trello issue](https://trello.com/c/4MDpUUc8/14-add-error-tracking-sentry)

## Pages/Interfaces that will change
All pages now have an `<ErrorBoundary />` wrapper

## Steps to test

1. `$ npm test`
2. `Look for ✅ `
3. `$ npm run:e2e:ci`
4. `Look for ✅ `

### Additional notes
This PR also adds an jest config to ignore all cypress specs 🤕

### CODE REVIEW

File: gatsby-config.js [hunk 1]
```diff
@@ -101,5 +101,14 @@ module.exports = {
       },
     },
     'gatsby-plugin-react-helmet',
+    {
+      resolve: 'gatsby-plugin-sentry',
+      options: {
+        dsn: process.env.SENTRY_DSN,
+        environment: process.env.NODE_ENV,
+        enabled: (() =>
+          ['production', 'staging'].indexOf(process.env.NODE_ENV) !== -1)(),
```
Comment (jakeleboeuf) [inline gatsby-config.js:110]: Should have confirmed this- But assuming we've set our env vars in Netlify as `production` and `staging`?

File: jest.config.js [hunk 2]
```diff
@@ -6,7 +6,12 @@ module.exports = {
     '.+\\.(css|styl|less|sass|scss)$': `identity-obj-proxy`,
     '.+\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$': `<rootDir>/__mocks__/file-mock.js`,
   },
-  testPathIgnorePatterns: [`node_modules`, `\\.cache`, `<rootDir>.*/public`],
+  testPathIgnorePatterns: [
+    `node_modules`,
+    `\\.cache`,
+    `<rootDir>.*/public`,
+    `cypress`,
```
Comment (jakeleboeuf) [inline jest.config.js:13]: 🚒

File: src/components/ErrorBoundary.jsx [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  // ? This is purely here for testing purposes
+  componentDidMount() {
+    this.props.forceError &&
+      logError({
+        error: new Error('Test errors'),
+        errorInfo: { componentStack: 'ErrorBoundary' },
+      });
+  }
```
Comment (jakeleboeuf) [inline src/components/ErrorBoundary.jsx:23]: Not sure how I feel about this, but was getting tired of looking for a reliable way to get jest to chill out and let me trigger errors... So I threw in a prop to simulate the behavior.

File: gatsby-config.js [hunk 4]
```diff
@@ -101,5 +101,14 @@ module.exports = {
       },
     },
     'gatsby-plugin-react-helmet',
+    {
+      resolve: 'gatsby-plugin-sentry',
+      options: {
+        dsn: process.env.SENTRY_DSN,
```
Comment (jakeleboeuf) [inline gatsby-config.js:107]: @magnificode @racedale @domitriusclark ☝️

File: jest.config.js (hunk 2 above)
Comment (racedale) [reply jest.config.js:13]: I also ran into needing to add this for CI. should cause a minor merge conflict with master

Comment (jakeleboeuf) [reply jest.config.js:13]: nice, I just cleaned up the conflicts 🚀

File: gatsby-config.js (hunk 4 above)
Comment (racedale) [reply gatsby-config.js:107]: I made a Sentry account in our 1Password should be able to go here and grab the DSN key
 https://sentry.io/settings/rebuild-black-business-m2/projects/rbb-website/keys/

Comment (magnificode) [reply gatsby-config.js:107]: DSN variable added to Netlify!

File: gatsby-config.js (hunk 1 above)
Comment (magnificode) [reply gatsby-config.js:110]: Yeah the env names are all default so production is what we've got on Netlify!

**Comment by magnificode** [comment]
@jakeleboeuf this looks solid to me, granted I don't have experience with Sentry or handling ErrorBoundaries :D - I'll leave to @racedale to approve and merge when he's comfortable

**Pull request merged**